### DISCUSSION TITLE
Improve web search when giving the AI a specific URL to crawl.

### DISCUSSION THREAD

**Discussion by mdillondc** [open-webui/open-webui#3997] [General]
**Is your feature request related to a problem? Please describe.**
chatgpt.com: Prompt `Is this product a solid choice? https://cdon.no/bygg-verktoy/watch-repair-tool-demagnetization-instrument-mechanical-watch-demagnetizer-style-blue-home-eu-plug-p153957184`

open-webui: Prompt `Is this product a solid choice? https://cdon.no/bygg-verktoy/watch-repair-tool-demagnetization-instrument-mechanical-watch-demagnetizer-style-blue-home-eu-plug-p153957184`

**Describe the solution you'd like**
ChatGPT will look up the URL provided and give feedback based on the specific product. 

open-webui will say "generating search query" and then go onto search various sites, often not including the one specified, and give a very general answer not really related to the product I specifically linked to.

Is it possible to have open-webui look only at the supplied URL and generate a response based on its contents instead of attempting to create a generic search prompt with whatever default search engine configured in settings?

**Comment by mdillondc** [comment]
I do realize... now, that I can do `#https://cdon.no/bygg-verktoy/watch-repair-tool-demagnetization-instrument-mechanical-watch-demagnetizer-style-blue-home-eu-plug-p153957184 ENTER` then ask my question, but still, it would be more user friendly if open-webui could figure that out on its own

**Comment by atgehrhardt** [comment]
This exists via the AutoTool Filter function on openwebui.com

You need a capable model, but as long as you have a decent model it can automatically search the web when needed.

**Reply by TomLucidor** [reply]
What about tool use for local model?

**Reply by atgehrhardt** [reply]
It depends on the model, but generally speaking I've been seeing tool calling work with newer small models.

Qwen2.5:14b has been quite exceptional at function calling and by extension tool calling for a smaller local model.

**Reply by TomLucidor** [reply]
@atgehrhardt are there models that are good at both tool use and coding?
